doc/diagnostics/WritingDdrCode.md

```diff
@@ -0,0 +1,174 @@
+<!--
+Copyright (c) 2021, 2021 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Writing Java code for DDR
+
+Data structures used by an active OpenJ9 JVM are described in C/C++ header files. Information
+about the fields of those data structures, their names, types and offsets are used by compilers
+to produce native code.
+
+That native code only needs to deal with a single version of those data structures - the
+one based on the source code at build time. DDR code, on the other hand, is used to navigate
+historical versions of data structures captured in system dumps from a VM built some time in
+the past. Historical versions may differ from the latest source code in a number of ways.
```

Comment:
I always wondered what is the main advantage for DDR being backwards compatible? Is it just ease of use, i.e. not having to download the same JVM that produced the core all the time, or is there more to it?

Reply:
Yes, but this is huge advantage. It creates universal kit might be used (included) in different tools as well as simplify debugging dramatically (cross platforms plus backward compatibility)

Reply:
I invite you to think of those people tasked with supporting customers and imagine how many versions of tools you would need to track and how much time you would waste when you start investigating a problem with the wrong version of your tools.

Reply:
And, as Dmitri points out, multiply that by the number of platforms you need to support.